src/cc/bcc_syms.cc

```diff
@@ -421,6 +508,45 @@ void bcc_symcache_refresh(void *resolver) {
   cache->refresh();
 }
 
+void *bcc_buildsymcache_new(void) {
+  return static_cast<void *>(new BuildSyms());
+}
+
+void bcc_free_buildsymcache(void *symcache) {
+  delete static_cast<BuildSyms*>(symcache);
+}
+
+int  bcc_buildsymcache_add_module(void *resolver, const char *module_name)
+{
+  BuildSyms *bsym = static_cast<BuildSyms *>(resolver);
+  return  bsym->add_module(module_name) ? 0 : -1;
+}
+
+int bcc_buildsymcache_resolve(void *resolver,
+                              struct bcc_stacktrace_build_id *trace,
+                              struct bcc_symbol *sym)
+{
+  std::string build_id;
+  unsigned char *c = &trace->build_id[0];
+  int idx = 0;
+
+  /*cannot resolve in case of fallback*/
+  if (trace->status == BCC_STACK_BUILD_ID_EMPTY or
+      trace->status == BCC_STACK_BUILD_ID_IP)
+    return 0;
+
+  while( idx < 20) {
+    int nib1 = (c[idx]&0xf0)>>4;
+    int nib2 = (c[idx]&0x0f);
+    build_id += "0123456789abcdef"[nib1];
+    build_id += "0123456789abcdef"[nib2];
+    idx++;
+  }
```

Comment:
Is the above loop for endianness? If this case, it probably won't work for bigendian.